File: 45. Jump Game II/Solution.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n,cnt,far,c = len(nums),0,0,0
        for i in range(n-1):
            far = max(far,i+nums[i])
            if i==c:
                cnt += 1
                c = far
        return cnt

File: 45. Jump Game II/Solution.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int i=0,cnt=0,far=0,c=0;
        while(i<n-1){
            far = max(far,i+nums[i]);
            if(i==c){
                cnt++;
                c = far;
            }
            i++;
        }
        return cnt;
    }
};
